src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -369,6 +377,16 @@ public Map<TopicPartition, OffsetAndMetadata> preCommit(
               committedOffsets.put(topicPartition, new OffsetAndMetadata(offSet));
             }
           });
+    } catch (SFException e) {
+      this.DYNAMIC_LOGGER.error("PreCommit SFException: {}", e.getMessage());
+
+      // if open channel failure, throw exception to kill kc
+      if (e.getVendorCode().equals(ErrorCode.OPEN_CHANNEL_FAILURE.getMessageCode())) {
```

Comment:
open channel failure could because of something transient so we would want to retry at least a few times?

Reply:
the retry should be handled lower down in the fallback api, though ill go through and verify that